onnxruntime/core/session/utils.cc

```diff
@@ -175,6 +178,13 @@ OrtStatus* CreateSessionAndLoadModel(_In_ const OrtSessionOptions* options,
   if (auto_select_ep_name) {
     ORT_API_RETURN_IF_STATUS_NOT_OK(TestAutoSelectEPsImpl(env->GetEnvironment(), *sess, *auto_select_ep_name));
   }
+
+  // if there are no providers registered, and there's an ep selection policy set, do auto ep selection
+  if (options->provider_factories.empty() &&
```

Comment:
I left the above path of explicit selection via 'test.ep_to_select' config value for now. Until the delegate is implemented that is the best way to test EP selection without the selection policy being used.